Client/WarFare/UIInventory.cpp

```diff
@@ -1494,16 +1495,34 @@ bool CUIInventory::ReceiveMessage(CN3UIBase* pSender, uint32_t dwMsg)
 					
 					// Get Item..
 					spItem = GetHighlightIconItem((CN3UIIcon* )pSender);
-
-					CN3UIWndBase::m_sSelectedIconInfo.UIWndSelect.UIWnd = UIWND_INVENTORY;
-					if (bSlot)
-						CN3UIWndBase::m_sSelectedIconInfo.UIWndSelect.UIWndDistrict = UIWND_DISTRICT_INVENTORY_SLOT;
-					else
-						CN3UIWndBase::m_sSelectedIconInfo.UIWndSelect.UIWndDistrict = UIWND_DISTRICT_INVENTORY_INV;
 					
+					CN3UIWndBase::m_sSelectedIconInfo.UIWndSelect.UIWnd = UIWND_INVENTORY;
 					CN3UIWndBase::m_sSelectedIconInfo.UIWndSelect.iOrder = iRBtn;
 					CN3UIWndBase::m_sSelectedIconInfo.pItemSelect = spItem;
-					
+					if (bSlot) //player clicked on equiped items
+					{
+						CN3UIWndBase::m_sSelectedIconInfo.UIWndSelect.UIWndDistrict = UIWND_DISTRICT_INVENTORY_SLOT;
+					}
+					else //player clicked on item which is not equiped
+					{
+						CN3UIWndBase::m_sSelectedIconInfo.UIWndSelect.UIWndDistrict = UIWND_DISTRICT_INVENTORY_INV;
+						
+						//right click on item to hotkey (skillbar)
+						if (spItem->pItemBasic->byClass == ITEM_CLASS_POTION) //usable items
```

Comment:
My biggest issue is here. You're assuming that these are the only type of items that can be moved to the skillbar.

Coincidentally, the very first thing I grabbed from Sundries was an Acid Potion -- which does *not* use this type, but *can* still be attached.

Rather than try to reinvent the wheel here with the logic, you should reuse/share the existing logic it uses to identify which items can be dragged onto the bar. Then it will behave consistently for both.